Lesson-5/assignment/src/components/EmployeeList.js

```diff
@@ -0,0 +1,208 @@
+import React, { Component } from 'react'
+import { Table, Button, Modal, Form, InputNumber, Input, message, Popconfirm } from 'antd';
+
+import EditableCell from './EditableCell';
+
+const FormItem = Form.Item;
+
+const columns = [{
+  title: '地址',
+  dataIndex: 'address',
+  key: 'address',
+}, {
+  title: '薪水',
+  dataIndex: 'salary',
+  key: 'salary',
+}, {
+  title: '上次支付',
+  dataIndex: 'lastPaidDay',
+  key: 'lastPaidDay',
+}, {
+  title: '操作',
+  dataIndex: '',
+  key: 'action'
+}];
+
+class EmployeeList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: true,
+      employees: [],
+      showModal: false
+    };
+
+    columns[1].render = (text, record) => (
+      <EditableCell
+        value={text}
+        onChange={ this.updateEmployee.bind(this, record.address) }
+      />
+    );
+
+    columns[3].render = (text, record) => (
+      <Popconfirm title="你确定删除吗?" onConfirm={() => this.removeEmployee(record.address)}>
+        <a href="#">Delete</a>
+      </Popconfirm>
+    );
+  }
+
+  componentDidMount() {
+    const { payroll, account, web3 } = this.props;
+    payroll.checkInfo.call({
+      from: account
+    }).then((result) => {
+      const employeeCount = result[2].toNumber();
+
+      if (employeeCount === 0) {
+        this.setState({loading: false});
+        return;
+      }
+
+      this.loadEmployees(employeeCount);
+    });
+
+  }
+
+  loadEmployees(employeeCount) {
+    const { payroll, account, web3 } = this.props;
+    const requests = [];
+    for ( let index = 0; index < employeeCount; index++) {
+      requests.push(payroll.checkEmployee.call(index, {
+        from: account
+      }))
+    }
+
+    Promise.all(requests
+    ).then(values => {
+      const employees = values.map(value => ({
+        key: value[0],
+        address: value[0],
+        salary: web3.fromWei(value[1].toNumber()),
+        lastPaidDay: new Date(value[2].toNumber() * 1000).toString()
+      }));
+
+      this.setState({
+        employees,
+        loading: false
+      })
+    })
+  }
+
+  addEmployee = () => {
+    const { payroll, account, web3 } = this.props;
+    const { address, salary, employees } = this.state;
+    payroll.addEmployee(address, salary, {
+      from: account,
+      gas: 300000
+    }).then(() => {
+      const newEmployee = {
+        address,
+        salary,
+        key: address,
+        lastPaidDay: new Date().toString()
```

Comment:
this is incorrect. You should read the real data from blockchain rather than fake it here.

the reason is `lastPaidDay` will be a timestamp the miner selected at his/her wish. So you don't know the exact number. The only way is to ask the blockchain to give you the data you need